app/controllers/api/user_blocks_controller.rb

```diff
@@ -14,5 +16,30 @@ def show
     rescue ActiveRecord::RecordNotFound
       raise OSM::APINotFoundError
     end
+
+    def create
+      raise OSM::APIBadUserInput, "No user was given" unless params[:user]
+
+      user = User.visible.find_by(:id => params[:user])
+      raise OSM::APINotFoundError unless user
+      raise OSM::APIBadUserInput, "No reason was given" unless params[:reason]
```

Comment:
side note: "Reason" doesn't seem to have a max size check in the model. Maybe it would be good to introduce an upper limit, like 10'000 chars or something similar.

Regarding the reason_format, we're assuming the default "markdown", and don't want to give API users a choice here, right?

Reply:
Neither of these comments has much to do with this PR which just introduces a new way of creating a block - both of those comments apply to the existing web interface as much as they do to an API call.

As far as format goes only markdown is supported for new data - the other formats are all legacy formats though I suspect they predate blocks in any case so all blocks are probably markdown.